lib/setup.js

```diff
@@ -0,0 +1,222 @@
+import path from 'path'
+import session from 'cookie-session'
+import sharify from 'sharify'
+import siteAssociation from 'artsy-eigen-web-association'
+import superSync from 'backbone-super-sync'
+import { IpFilter as ipfilter } from 'express-ipfilter'
+import './setup_sharify'
+import cache from './cache'
+import downcase from './middleware/downcase'
+import ensureSSL from './middleware/ensure_ssl'
+import escapedFragmentMiddleware from './middleware/escaped_fragment'
+import flashMiddleware from './middleware/flash'
+import hardcodedRedirects from './middleware/hardcoded_redirects'
+import hstsMiddleware from './middleware/hsts'
+import localsMiddleware from './middleware/locals'
+import proxyGravity from './middleware/proxy_to_gravity'
+import proxyReflection from './middleware/proxy_to_reflection'
+import sameOriginMiddleware from './middleware/same_origin'
+import unsupportedBrowserCheck from './middleware/unsupported_browser'
+import logFormat from './logger_format'
+import CurrentUser from './current_user'
+import splitTestMiddleware from '../desktop/components/split_test/middleware'
+import config from '../config'
+
+const {
+  API_REQUEST_TIMEOUT,
+  API_URL,
+  COOKIE_DOMAIN,
+  DEFAULT_CACHE_TIME,
+  IP_BLACKLIST,
+  NODE_ENV,
+  OPENREDIS_URL,
+  REQUEST_EXPIRE_MS,
+  REQUEST_LIMIT,
+  SESSION_COOKIE_KEY,
+  SESSION_COOKIE_MAX_AGE,
+  SESSION_SECRET
+} = config
+
+export default function (app) {
+  // Blacklist IPs
+  app.use(ipfilter(IP_BLACKLIST.split(','), { log: false, mode: 'deny' }))
+
+  // Rate limiting
+  if (OPENREDIS_URL && cache.client) {
+    const limiter = require('express-limiter')(app, cache.client)
+    limiter({
+      path: '*',
+      method: 'all',
+      lookup: ['headers.x-forwarded-for'],
+      total: REQUEST_LIMIT,
+      expire: REQUEST_EXPIRE_MS,
+      onRateLimited (req, res, next) {
+        console.log('Rate limit exceeded for', req.headers['x-forwarded-for'])
+        return next()
+      }
+    })
+  }
+
+  // Blank page used by Eigen for caching web views.
+  // See: https://github.com/artsy/microgravity-private/pull/1138
+  app.use(require('../desktop/apps/blank'))
+
+  // Make sure we're using SSL and prevent clickjacking
+  app.use(ensureSSL)
+  app.use(hstsMiddleware)
+  if (!NODE_ENV === 'test') app.use(helmet.frameguard())
+
+  // Override Backbone to use server-side sync, inject the XAPP token,
+  // add redis caching, and timeout for slow responses.
+  superSync.timeout = API_REQUEST_TIMEOUT
+  superSync.cacheClient = cache.client
+  superSync.defaultCacheTime = DEFAULT_CACHE_TIME
+  Backbone.sync = function (method, model, options) {
+    if (options.headers == null) { options.headers = {} }
+    options.headers['X-XAPP-TOKEN'] = artsyXapp.token || ''
+    return superSync(method, model, options)
+  }
+
+  // Inject sharify data before any app code
+  app.use(sharify)
+
+  // Add stubbed API servers and a stubbed user for testing
+  // TODO: https://github.com/artsy/force/issues/1149
```

Comment:
I actually completed this TODO, so I need to remove this part where we mount fake gravity/positron servers for testing. Updating...